### ISSUE TITLE
Commit "Fix side effect when referencing the same resource in multiple stacks" breaks deployment

### ISSUE THREAD

**Issue by carlcheel-sage** [dougmoscrop/serverless-plugin-split-stacks#117]
Following the addition of commit [8273717...](https://github.com/dougmoscrop/serverless-plugin-split-stacks/commit/827371789a07144977d59de4690b73a3fb777dfb) the `ServerlessDeploymentBucketPolicy` has the incorrect ref when building.

Before commit `827371789a07144977d59de4690b73a3fb777dfb` (works fine):

```
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "PoliciesNestedStack nested stack",
  "Parameters": {
    "ServerlessDeploymentBucketParameter": {
      "Type": "String"
    }
  },
  "Resources": {
    "ServerlessDeploymentBucketPolicy": {
      "Type": "AWS::S3::BucketPolicy",
      "Properties": {
        "Bucket": {
          "Ref": "ServerlessDeploymentBucketParameter"
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Effect": "Deny",
              "Principal": "*",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition"
                      },
                      ":s3:::",
                      {
                        "Ref": "ServerlessDeploymentBucketParameter"
                      },
                      "/*"
                    ]
                  ]
                }
              ],
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": false
                }
              }
            }
          ]
        }
      },
      "DependsOn": []
    }
  },
  "Outputs": {}
}
```

After (broken due to incorrect Ref):

```
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "PoliciesNestedStack nested stack",
  "Parameters": {
    "ServerlessDeploymentBucketParameter": {
      "Type": "String"
    }
  },
  "Resources": {
    "ServerlessDeploymentBucketPolicy": {
      "Type": "AWS::S3::BucketPolicy",
      "Properties": {
        "Bucket": {
          "Ref": "ServerlessDeploymentBucket"
        },
        "PolicyDocument": {
          "Statement": [
            {
              "Action": "s3:*",
              "Effect": "Deny",
              "Principal": "*",
              "Resource": [
                {
                  "Fn::Join": [
                    "",
                    [
                      "arn:",
                      {
                        "Ref": "AWS::Partition"
                      },
                      ":s3:::",
                      {
                        "Ref": "ServerlessDeploymentBucket"
                      },
                      "/*"
                    ]
                  ]
                }
              ],
              "Condition": {
                "Bool": {
                  "aws:SecureTransport": false
                }
              }
            }
          ]
        }
      },
      "DependsOn": []
    }
  },
  "Outputs": {}
}
```

**Comment by carlcheel-sage** [comment]
Confirmed the issue is caused by `_cloneDeep` on [Line 234](https://github.com/dougmoscrop/serverless-plugin-split-stacks/commit/827371789a07144977d59de4690b73a3fb777dfb#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR234) and [Line 239](https://github.com/dougmoscrop/serverless-plugin-split-stacks/commit/827371789a07144977d59de4690b73a3fb777dfb#diff-50e3aa130a4f97a42ee2cf111c7b1d9dR239). Removing this fixes the issue.

**Comment by carlcheel-sage** [comment]
Closed as duplicate of issue #82
